src/rust/engine/src/fs.rs

```diff
@@ -636,26 +635,26 @@ impl fmt::Debug for Snapshot {
 }
 
 /**
- * A facade for the snapshot directory, which is currently thrown away at the end of each run.
+ * A facade for the snapshot directory, which lives under the pants workdir.
  */
 pub struct Snapshots {
-  temp_dir: TempDir,
+  snapshots_dir: PathBuf,
   next_temp_id: atomic::AtomicUsize,
 }
 
 impl Snapshots {
-  pub fn new() -> Result<Snapshots, io::Error> {
+  pub fn new(snapshots_dir: PathBuf) -> Result<Snapshots, io::Error> {
+    fs::create_dir_all(snapshots_dir.to_owned())?;
```

Comment:
ah, makes sense. `.as_path()` worked. dropping the `.to_owned()` gave me:

```
error[E0382]: use of moved value: `snapshots_dir`
   --> src/rust/engine/src/fs.rs:650:24
    |
647 |     fs::create_dir_all(snapshots_dir)?;
    |                        ------------- value moved here
...
650 |         snapshots_dir: snapshots_dir,
    |                        ^^^^^^^^^^^^^ value used here after move
    |
    = note: move occurs because `snapshots_dir` has type `std::path::PathBuf`, which does not implement the `Copy` trait
```

which I saw initially and had went with `.to_owned()` essentially as a higher level alternative for `.clone()`.